test/e2e/hot-reload.test.js

```diff
@@ -30,29 +33,26 @@ describe('hot-reload', () => {
     let testGatewayConfigData = null;
     let childProcess = null;
     let originalGatewayPort = null;
+    let watcher = null;
 
     before(function (done) {
-      this.timeout(10000);
+      this.timeout(TEST_TIMEOUT);
       tmp.dir((err, tempPath) => {
         if (err) {
-          throw err;
+          return done(err);
         }
 
         cpr(baseConfigDirectory, tempPath, (err, files) => {
           if (err) {
-            throw err;
+            return done(err);
```

Comment:
For the future generations: please never throw an exception from an async flow. Nobody can catch the value and the current process will crash, entirely.